### PR TITLE
refactor: use records

### PR DESCRIPTION
**Pull request by italo1aguiar-aneo**

### CODE REVIEW

File: ArmoniK.Extension.CSharp.Client/Common/Domain/Task/TaskPagination.cs [hunk 1]
```diff
@@ -27,3 +27,41 @@ public class TaskPagination
...
+  /// <summary>
+  ///   Total number of pages available in the paginated list.
+  /// </summary>
+  public int TotalPages { get; init; }
+
+  /// <summary>
+  ///   Detailed state information of the task.
+  /// </summary>
+  public TaskState TaskDetails { get; init; }
+}
```
Comment (aneojgurhem) [inline ArmoniK.Extension.CSharp.Client/Common/Domain/Task/TaskPagination.cs:67]: that is not a page, it is only one value. I'd expect an array

File: ArmoniK.Extension.CSharp.Client/Common/Domain/Task/TaskPagination.cs [hunk 2]
```diff
@@ -27,3 +27,41 @@ public class TaskPagination
   public SortDirection SortDirection { get; set; }
   public Filters       Filter        { get; set; }
 }
+
+/// <summary>
+///   Represents a page within a paginated list of tasks, providing basic task status information.
+/// </summary>
+public record TaskPage
```
Comment (aneojgurhem) [inline ArmoniK.Extension.CSharp.Client/Common/Domain/Task/TaskPagination.cs:34]: Pagination is the same everywhere. I think some code can be factorized

File: ArmoniK.Extension.CSharp.Client/Common/Domain/ApplicationConfiguration.cs [hunk 3]
```diff
@@ -16,11 +16,33 @@
 
 namespace ArmoniK.Extension.CSharp.Client.Common.Domain;
```
Comment (aneojgurhem) [inline ArmoniK.Extension.CSharp.Client/Common/Domain/ApplicationConfiguration.cs:17]: IMO, this should be part of the worker associated extensions

Comment (italo1aguiar-aneo) [reply ArmoniK.Extension.CSharp.Client/Common/Domain/ApplicationConfiguration.cs]: I agree, I'll remove this class.

File: UsageExample/Program.cs [hunk 4]
```diff
@@ -32,8 +32,8 @@ namespace UsageExample;
 
 internal class Program
 {
-  private static IConfiguration   _configuration;
-  private static ILogger<Program> logger_;
+  private static IConfiguration?   _configuration;
+  private static ILogger<Program>? _logger;
```
Comment (aneojgurhem) [inline UsageExample/Program.cs:36]: these changes should not happen. Formating expects the previous version.

File: ArmoniK.Extension.CSharp.Client/Common/Domain/Blob/BlobState.cs [hunk 5]
```diff
@@ -20,28 +20,91 @@
...
+
+  /// <summary>
+  ///   Datetime when the blob was created.
+  /// </summary>
+  public DateTime? CreateAt { get; init; }
+
+  /// <summary>
+  ///   Current status of the blob.
+  /// </summary>
+  public BlobStatus? Status { get; init; }
```
Comment (aneojgurhem) [inline ArmoniK.Extension.CSharp.Client/Common/Domain/Blob/BlobState.cs:41]: those 2 should not be null

File: ArmoniK.Extension.CSharp.Client/Common/Domain/Task/TaskState.cs [hunk 6]
```diff
@@ -16,34 +16,178 @@
...
+
+  /// <summary>
+  ///   Time when the task ended.
+  /// </summary>
+  public DateTime? EndedAt { get; init; }
+
+  /// <summary>
+  ///   Time when the task started.
+  /// </summary>
+  public DateTime StartedAt { get; init; }
```
Comment (aneojgurhem) [inline ArmoniK.Extension.CSharp.Client/Common/Domain/Task/TaskState.cs:66]: this one can be null if task did not start

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**